executor/src/witgen/processor.rs

```diff
@@ -639,6 +640,24 @@ Known values in current row (local: {row_index}, global {global_row_index}):
             ),
         };
 
+        if let Ok(connection) = Connection::try_from(identity) {
+            // JITed submachines would panic if passed a wrong input / output pair.
+            // Therefore, if any machine call is activated, we resort to the full
+            // solving routine.
+            // An to this is when the call is always active (e.g. the PC lookup).
+            // In that case, we know that the call has been active before with the
+            // same input / output pair, so we can be sure that it will succeed.
```

Comment:
A bit of a hack, but I think we'll get rid of the loop detection business soon anyway. We could even consider removing it already, since dynamic VADCOP should make sure that we don't waste a lot of time after the main function returned.